frappe/core/doctype/domain_settings/domain_settings.py

```diff
@@ -3,20 +3,18 @@
 # For license information, please see license.txt
 
 from __future__ import unicode_literals
-import frappe
+import frappe, json
 from frappe.model.document import Document
 from frappe import _
 
 class DomainSettings(Document):
 	def set_active_domains(self, domains):
-		self.active_domains = []
-		for d in domains:
-			self.append('active_domains', dict(domain=d))
+		self.active_domains = json.dumps(domains);
 		self.save()
 
 	def on_update(self):
-		for d in self.active_domains:
-			domain = frappe.get_doc('Domain', d.domain)
+		for d in json.loads(self.active_domains):
```

Comment:
this should be handled in client side. server has nothing to do with controls!

Reply:
So maybe we could parse Multi Check control values as `list` by default ?

Reply:
Yes, had tried to set their value as a list itself, but wasn't sure how we'd store the field value in the database then.